src/Symfony/Component/Console/Output/ConsoleSectionOutput.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Output;
+
+use Symfony\Component\Console\Formatter\OutputFormatterInterface;
+use Symfony\Component\Console\Terminal;
+
+/**
+ * @author Pierre du Plessis <pdples@gmail.com>
+ */
+class ConsoleSectionOutput extends StreamOutput
+{
+    public $content;
+    public $lines = 0;
+
+    private $sectionReference;
+    private $terminal;
+
+    /**
+     * @param resource                 $stream
+     * @param OutputSectionReference   $sectionReference
+     * @param int                      $verbosity
+     * @param null                     $decorated
+     * @param OutputFormatterInterface $formatter
+     */
+    public function __construct($stream, OutputSectionReference $sectionReference, $verbosity, $decorated, OutputFormatterInterface $formatter)
```

Comment:
`..., $verbosity = self::VERBOSITY_NORMAL, $decorated = null, OutputFormatterInterface $formatter = null`?

Reply:
I'm not sure if these values should to be defaulted, as we want to inherit all the values from origin console output

Reply:
It's pragmatic right now, out-the-box we would want to use defaults.

Reply:
This class should never be used on it's own, it's just an implementation detail and should always be constructed from the `ConsoleOutput`, which means it should always inherit the values set there (default or otherwise)

Reply:
That means the PHPDoc of the constructor could be removed altogether if so?

Reply:
Dockblock removed